### PR TITLE
enhance: add trace_id into segcore logs

### PR DESCRIPTION
**Pull request by chyezh**
issue: #37655

### CODE REVIEW

File: internal/core/src/common/Tracer.cpp [hunk 1]
```diff
@@ -134,6 +134,28 @@ StartSpan(const std::string& name, const std::shared_ptr<trace::Span>& span) {
...
+    }
+    auto ctx = local_span->GetContext();
+    auto trace_id = ctx.trace_id();
+    // The span is noop if the trace is off, 00000000... will be returned,
+    // so return "0" string directly to distinguish from the no local span.
+    if (!trace_id.IsValid()) {
+        return "0";
+    }
+    auto rep = trace_id.Id();
+    return BytesToHexStr(rep.data(), rep.size());
```
Comment (congqixia) [inline internal/core/src/common/Tracer.cpp:156]: will the trace str will be calculated each log call? can this be avoided?

**Review by congqixia — COMMENTED** [review]
overall lgtm
just a small comment

File: internal/core/src/common/Tracer.cpp (hunk 1 above)
Comment (chyezh) [reply internal/core/src/common/Tracer.cpp:156]: yes, save the context at root will fix it. 
But in our current implementation of segcore, LOG is low rate oepartion when working on INFO level at segcore, so make a pre-calculation will generate make-no-sense opeartion for most search opeartion, should not be a good-choice.
At DEBUG level, we don't care the performance of such a string building.

Comment (congqixia) [reply internal/core/src/common/Tracer.cpp:156]: Maybe a lazy+pre calc could be done, could be optimized in future lgtm now

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**